agent/ebs/watcher.go

```diff
@@ -69,6 +88,9 @@ func (w *EBSWatcher) Start() {
 			if len(pendingEBS) > 0 {
 				foundVolumes := apiebs.ScanEBSVolumes(pendingEBS, w.discoveryClient)
 				w.overrideDeviceName(foundVolumes)
+				if err := w.StageAll(foundVolumes); err != nil {
+					log.Errorf("stage error: %s", err)
```

Comment:
We should probably want to skip this tick iteration if we do encounter an error. Perhaps we can add a `continue` here.

Reply:
(non-blocking question/idea! feel free to ignore if this isn't a good idea/takes too much) thinking about it now, would it also make sense if we return a list of volume IDs that were staged successfully and then pass it onto NotifyAttached? In that case we'd would log that there was a stage error for specific volumes instead if there were any errors overall (might also make it a bit easier to debug)

Reply:
another thought would be to let the attachments themselves handle this.  we update the attachment status only after successful stage and we don't send unless the attachment status is marked attached.

Reply:
Yeah I think we should process notify for the ones that succeeded.

Reply:
added the continue and a TODO to update stage all to return only successfully staged.